### PR TITLE
ENT-8983 Upgrade H2 to latest version

### PR DESCRIPTION
**Pull request by mahmoudal993**
* Updated H2 to latest version 2.1.212

### CODE REVIEW

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/MockServices.kt [hunk 1]
```diff
@@ -114,7 +114,7 @@ open class MockServices private constructor(
             }
             val props = Properties()
             props.setProperty("dataSourceClassName", "org.h2.jdbcx.JdbcDataSource")
-            props.setProperty("dataSource.url", "jdbc:h2:file:$dbPath;LOCK_TIMEOUT=10000;DB_CLOSE_ON_EXIT=FALSE")
+            props.setProperty("dataSource.url", "jdbc:h2:mem:$dbPath;NON_KEYWORDS=KEY,VALUE;LOCK_TIMEOUT=10000;DB_CLOSE_ON_EXIT=FALSE")
```
Comment (rick-r3) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/MockServices.kt]: Why did you change this from file to mem ?

Comment (mahmoudal993) [reply testing/node-driver/src/main/kotlin/net/corda/testing/node/MockServices.kt]: I had temporarily changed that when I first started working on this and I was testing different things I forgot to revert it back to file.

**Review by chriscochrane — COMMENTED** [review]
I presume that all the changes in SchemaMigration::CustomResourceAccessor basically provide the same contract as the previous code, and that the old getresourceAsStream() method has been superceded by getAll().
Code changes all look OK to me.

**Review by chriscochrane — APPROVED** [review]

**Review by rick-r3 — APPROVED** [review]

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/MockServices.kt [hunk 2]
```diff
@@ -114,7 +114,7 @@ open class MockServices private constructor(
             }
             val props = Properties()
             props.setProperty("dataSourceClassName", "org.h2.jdbcx.JdbcDataSource")
-            props.setProperty("dataSource.url", "jdbc:h2:file:$dbPath;LOCK_TIMEOUT=10000;DB_CLOSE_ON_EXIT=FALSE")
+            props.setProperty("dataSource.url", "jdbc:h2:file:$dbPath;NON_KEYWORDS=KEY,VALUE;LOCK_TIMEOUT=10000;DB_CLOSE_ON_EXIT=FALSE")
```
Comment (shamsasari) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/MockServices.kt:117]: What is the `NON_KEYWORDS=KEY,VALUE` statement doing?

Comment (mahmoudal993) [reply testing/node-driver/src/main/kotlin/net/corda/testing/node/MockServices.kt:117]: In the latest h2 version "KEY" and "VALUE" are flagged as keywords and throws an error. To avoid making changes to our tables/columns we set these as NON_KEYWORDS.

Comment (mahmoudal993) [reply testing/node-driver/src/main/kotlin/net/corda/testing/node/MockServices.kt:117]: https://www.h2database.com/html/commands.html#set_non_keywords

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/persistence/SchemaMigration.kt [hunk 3]
```diff
@@ -46,31 +56,31 @@ open class SchemaMigration(
...
-                     runner.update(Contexts().toString())
-                 } catch (exp: LiquibaseException) {
-                     throw DatabaseMigrationException(exp.message, exp)
-                 }
-             }
-         }
-     }
+    fun runMigration(existingCheckpoints: Boolean, schemas: Set<MappedSchema>, forceThrowOnMissingMigration: Boolean) {
+        val resourcesAndSourceInfo = prepareResources(schemas, forceThrowOnMissingMigration)
+        Scope.enter(mapOf(Scope.Attr.classLoader.name to classLoader))
```
Comment (shamsasari) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/persistence/SchemaMigration.kt:68]: Should there not be a corresponding `Scope.exit` at the end?
